File: src/main/java/com/backend/immilog/image/application/service/query/ImageQueryService.java
package com.backend.immilog.image.application.service.query;

import com.backend.immilog.image.domain.model.Image;
import com.backend.immilog.image.domain.repository.ImageRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class ImageQueryService {
    private final ImageRepository imageRepository;

    public ImageQueryService(ImageRepository imageRepository) {
        this.imageRepository = imageRepository;
    }

    @Transactional(readOnly = true)
    public Image getImageByPath(String imagePath) {
        return imageRepository.findByPath(imagePath);
    }
}

File: src/main/java/com/backend/immilog/image/application/service/query/ImageQueryService.kt
package com.backend.immilog.image.application.service.query

import com.backend.immilog.image.domain.model.Image
import com.backend.immilog.image.domain.repository.ImageRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class ImageQueryService(
    private val imageRepository: ImageRepository
) {
    @Transactional(readOnly = true)
    fun getImageByPath(imagePath: String): Image = imageRepository.findByPath(imagePath)
}
